.pre-commit-config.yaml

```diff
@@ -40,5 +40,13 @@ repos:
       - id: poetry-check
         args: ['-C', 'api']
 
+  - repo: local
+    hooks:
+      - id: mypy-check
+        name: mypy check
+        entry: bash -c "cd api && poetry run python scripts/mypy_check.py $(git diff --name-only --cached -- '*.py')"
```

Comment:
As seen from pre-commit.ci run [here](https://results.pre-commit.ci/run/github/136163130/1733506020.vLWIKPvMRNi862KNAN5dkA), this hook won't work when poetry isn't present globally.

Maybe try adding

```
        additional_dependencies: ['poetry']
```

or using `repo: local` (see docs [here](https://pre-commit.com/#repository-local-hooks)).

If all else fails, moving the hook to its own repo could be an option.

Reply:
The `additional_dependencies` seems to have worked once I changed the `language` to `python`.